### ISSUE TITLE
Testnet block storm resiliency

### ISSUE THREAD

**Issue by kantai** [stacks-network/stacks-blockchain#3173]
For the most part, any issues experienced in the Stacks testnet these days are caused by Bitcoin testnet block storms. During these events, Stacks block production grinds to a halt. I'm starting this issue as a place to discuss feasible ideas for addressing this. I suspect that there's not _much_ that can be done, because testnet block storms typically have a huge number of empty blocks (which won't include a Stacks block commit in any event), but it may be possible to get occasional block commits included during the storm.

**Comment by CharlieC3** [comment]
Jude mentioned in the weekly blockchain meeting that there may be a few configurations we can tweak for the stacks-node miner to reduce stalling during a bitcoin blockstorm. IIRC, those suggestions were changing the time to first block to something lower, like 5 seconds, and updating a config to allow microblocks to pack fewer TXs.

@jcnelson are you able to post exactly what config changes you're suggesting in the miner's `Config.toml`?

**Comment by jcnelson** [comment]
Can you try these?

```
[miner]
first_attempt_time_ms = 5000
subsequent_attempt_time_ms = 15000
microblock_attempt_time_ms = 5000

[node]
wait_time_for_microblocks = 0
```

**Comment by CharlieC3** [comment]
Thanks, I've applied these config changes. Let's see how it performs during the next BTC block storm.

**Comment by CharlieC3** [comment]
There's a tBTC storm occurring right now, and our testnet miner hasn't produced a block since the storm started about 16 hours ago. So it seems the above settings doesn't have much of an effect.

**Comment by igorsyl** [comment]
> ms the above settings doesn't have much of an effect.

Ditto here.

**Comment by jcnelson** [comment]
How often are testnet blocks arriving?  Can you at least make the timeouts in `[miner]` smaller than this value?

**Comment by CharlieC3** [comment]
@jcnelson On average it seems about one tBTC/minute ([checking here](https://www.blockchain.com/btc-testnet/blocks?page=1)). The timeouts listed [here](https://github.com/stacks-network/stacks-blockchain/issues/3173#issuecomment-1167675840) should already be much smaller than the number of tBTC blocks being produced.

**Comment by jcnelson** [comment]
Can you get me access to the logs?

**Comment by igorsyl** [comment]
> Can you get me access to the logs? As well as the full config?

Test miner
Config: https://gist.github.com/igorsyl/8af7426434d011255c7c1bb10bcd32f7
Logs: https://gist.github.com/igorsyl/81d8fa748e01d975115d2e5c724fe396

**Comment by jcnelson** [comment]
Thanks @igorsyl!

From the logs, you can see that it's _definitely_ mining -- it's submitting a block-commit every testnet block.  But, about half the time, the block-commit is invalid because the burn modulus is off-by-one (meaning the block-commit didn't make it into the targeted Bitcoin block, but into the one after it).

**Comment by jcnelson** [comment]
Maybe try bumping the tBTC fee?

**Comment by igorsyl** [comment]
Yes, it's burning through all of my hard earned testnet BTC! I doubled the tBTC fee and let's see what happens.

**Comment by jcnelson** [comment]
Keep in mind that it may do nothing, because the tBTC miner creating the blockstorm might not even be considering our transactions for reasons unknown to us.

**Comment by igorsyl** [comment]
I suspect this storm could be a way for someone to top-up their tBTC wallet.
What btc and stx config files in the repo would allow for a separate and private testnet?

**Comment by CharlieC3** [comment]
@igorsyl Not sure if this would help your situation, but you can spawn your own local devnet using [Clarinet](https://github.com/hirosystems/clarinet#spawn-a-local-devnet).

```
brew install clarinet
clarinet new my-project && cd my-project
clarinet integrate
```

Should be that easy

**Comment by igorsyl** [comment]
Thanks @CharlieC3! We are testing Bitcoin and Stacks transactions together so we need a testnet full-stack setup.

**Comment by jcnelson** [comment]
Temporarily assigning to @kantai so it has an owner.  Feel free to re-assign, close, or refactor.

The 2.05.0.6.0 release had a lot of changes to be able to deal with Stacks and Bitcoin tips changing while the node was mining.  Have we seen block storms degrade the chain quality with this release?

**Comment by kantai** [comment]
I'm iceboxing this for now
